src/EditorFeatures/Core/LanguageServer/Handlers/CodeActions/CodeActionsHandler.cs

```diff
@@ -59,24 +61,11 @@ public async Task<LSP.CodeAction[]> HandleRequestAsync(LSP.CodeActionParams requ
 
             var options = _globalOptions.GetCodeActionOptionsProvider();
 
+            var codeActionsCache = context.GetRequiredLspService<CodeActionsCache>();
```

Comment:
If we get the same codeActionsCache back each time this is called does it maybe make more sense to just hold onto it after the first time instead of constantly getting the service again? I'll admit this is cleaner but I wonder if the overhead of GetRequiredLspService calls each time will matter.

Reply:
I don't think it particularly matters either way - doing it this way reduces the boiler plate a bit and the handler can become stateless.  At the end of the day it is just a lookup into a map so there isn't really much cost here. 

If I store it on the type, the handler has to become stateful which is fine, but just more code.